internal/database/authorized_app.go

```diff
@@ -25,10 +25,20 @@ import (
 	pgx "github.com/jackc/pgx/v4"
 )
 
+type AuthorizedAppDB struct {
+	db *DB
+}
+
+func NewAuthorizedAppDB(db *DB) *AuthorizedAppDB {
+	return &AuthorizedAppDB{
+		db: db,
+	}
+}
+
 // GetAuthorizedApp loads a single AuthorizedApp for the given name. If no row
 // exists, this returns nil.
-func (db *DB) GetAuthorizedApp(ctx context.Context, sm secrets.SecretManager, name string) (*AuthorizedApp, error) {
-	conn, err := db.pool.Acquire(ctx)
+func (db *AuthorizedAppDB) GetAuthorizedApp(ctx context.Context, sm secrets.SecretManager, name string) (*AuthorizedApp, error) {
```

Comment:
it seems it may be a convention in Go to call these 'Repositories'. So to avoid the double `db.db` I could name this `repo *AuthorizedAppRepository`.

Reply:
"Repo" is already overloaded. Can we still call them models? `AuthorizedAppModel`?

Reply:
PTAL.